packages/editor/src/store/selectors.js

```diff
@@ -485,18 +489,29 @@ export function isEditedPostEmpty( state ) {
 /**
  * Returns true if the post can be autosaved, or false otherwise.
  *
- * @param  {Object}  state Global application state.
+ * @param  {Object} state    Global application state.
+ * @param  {Object} autosave An autosave object.
  *
  * @return {boolean} Whether the post can be autosaved.
  */
-export function isEditedPostAutosaveable( state ) {
+export function isEditedPostAutosaveable( state, autosave ) {
+	if ( arguments.length === 1 ) {
+		deprecated( '`wp.data.select( \'core/editor\' ).isEditedPostAutosaveable()`', {
```

Comment:
I'm not sure if this is the right way to handle a change of function signature.

Reply:
> I'm not sure if this is the right way to handle a change of function signature.

I think it's as good as we'll get with deprecation, for the moment.

The main issue here is that the top-level `select` function is not contextualized to a specific data registry, so it's not guaranteed to be operating on the same data as where the state provided to the selector is actually coming from. This is normally provided by `withSelect` using a `RegistryConsumer`:

https://github.com/WordPress/gutenberg/blob/71be7e60faac03624d37d436e91a258b72457131/packages/data/src/components/with-select/index.js#L149-L156

Previously I'd thought we'd want a way to be able to compose selectors across stores, though the idea of having the requirement be to pass data as arguments is also compelling. Selecting across stores is not simple to implement, and most likely wouldn't qualify as being considered "pure".

Reply:
Andrew is right here that using `select()` will always use the default registry and not any specific registry that is set using `RegistryProvider`. 

@aduth: do you have any suggestions on how to address this?

Reply:
I've updated the code to use registrySelectors in https://github.com/WordPress/gutenberg/pull/7945/commits/1a489e2b2dc2d545da3ad6c886e26166f9dd7503, the tests for `isEditedPostAutosaveable` also needed to be changed in https://github.com/WordPress/gutenberg/pull/7945/commits/e349caaab00b0a132cf4820a5b9988aad2ddc469 to take into account that it's a registry selector. I've added comments that hopefully make the situation clear.

Would also note that this PR introduces some use of `select` in an effect, and I haven't converted that to a registry selector. Not entirely sure of the best strategy there, but I note that there's also some existing use of select in effects:
https://github.com/WordPress/gutenberg/blob/master/packages/edit-post/src/store/effects.js#L46-L54

I've tested, the selectors still work, but unsure of how to assert that they work in the context of a different registry 🤔